src/infrastructure/persistence/PsqlMagicLinkRepository.ts

```diff
@@ -0,0 +1,51 @@
+import knex from 'knex';
+import { MagicLinkRepository } from '../../domain/model/magiclink/MagicLinkRepository';
+import { MagicLink } from '../../domain/model/magiclink/MagicLink';
+import { Email } from '../../domain/model/user/Email';
+
+export class PsqlMagicLinkRepository implements MagicLinkRepository {
+  constructor(private db: knex) {}
+
+  async findByCode(code: string): Promise<MagicLink | undefined> {
+    const magicLinkRow = await this.db('magic_link')
+      .where('code', '=', code)
+      .select([
+        'id',
+        'email',
+        'code',
+        'active',
+        'creation_time as creationTime',
+      ])
+      .first();
+    if (!magicLinkRow) {
+      return undefined;
```

Comment:
why undefined instead of null? It's not really undefined...

Reply:
good point.